src/azure-cli/azure/cli/command_modules/backup/custom_wl.py

```diff
@@ -320,8 +324,90 @@ def list_protectable_items(client, resource_group_name, vault_name, workload_typ
     return paged_items
 
 
+def fetch_tier(paged_recovery_points):
+
+    for rp in paged_recovery_points:
+        isRehydrated = False
+        isInstantRecoverable = False
+        isHardenedRP = False
+        isArchived = False
+
+        if rp.properties.recovery_point_tier_details is None:
+            continue
+
+        for i in range(len(rp.properties.recovery_point_tier_details)):
+            if (rp.properties.recovery_point_tier_details[i].type == "ArchivedRP" and
+                    rp.properties.recovery_point_tier_details[i].status == "Rehydrated"):
+                isRehydrated = True
+
+            if rp.properties.recovery_point_tier_details[i].status == "Valid":
+                if rp.properties.recovery_point_tier_details[i].type == "InstantRP":
+                    isInstantRecoverable = True
+
+                if rp.properties.recovery_point_tier_details[i].type == "HardenedRP":
+                    isHardenedRP = True
+
+                if rp.properties.recovery_point_tier_details[i].type == "ArchivedRP":
+                    isArchived = True
+
+        if (isHardenedRP and isArchived) or (isRehydrated):
+            setattr(rp, "tier_type", "VaultStandardRehydrated")
+
+        elif isInstantRecoverable and isHardenedRP:
+            setattr(rp, "tier_type", "SnapshotAndVaultStandard")
+
+        elif isInstantRecoverable and isArchived:
+            setattr(rp, "tier_type", "SnapshotAndVaultArchive")
+
+        elif isArchived:
+            setattr(rp, "tier_type", "VaultArchive")
+
+        elif isInstantRecoverable:
+            setattr(rp, "tier_type", "Snapshot")
+
+        elif isHardenedRP:
+            setattr(rp, "tier_type", "VaultStandard")
+
+
+def check_rp_move_readiness(paged_recovery_points, target_tier, is_ready_for_move):
```

Comment:
private function naming should start with underscore.

Reply:
done